specifyweb/frontend/js_src/lib/components/Interactions/InteractionDialog.tsx

```diff
@@ -238,6 +249,82 @@ export function InteractionDialog({
     );
   };
 
+  const [unavailableCatNumbers, setUnavailableCatNumbers] =
+    React.useState<RA<string>>();
+  const [availableCatNumbers, setAvailableCatNumbers] =
+    React.useState<RA<string>>();
+  const [availableData, setAvailableData] = React.useState<
+    | RA<{
+        readonly catalognumber: string;
+        readonly id: number;
+      }>
+    | undefined
+  >();
+
+  const hanleAvailableCatNumber = (): void => {
+    const catalogNumbers = handleParse();
+    if (catalogNumbers === undefined) return undefined;
+
+    loading(
+      (catalogNumbers.length === 0
+        ? Promise.resolve([])
+        : // Add await for catNum promise
+          getCatNumberAvailableForAccession('CatalogNumber', catalogNumbers)
+      )
+        .then((data) => {
+          setAvailableData(data);
+
+          const returnCOCatNumber = new Set(
+            data.map((item) => item.catalognumber)
+          );
+
+          const availableCOs = catalogNumbers.filter((catNumber) =>
+            returnCOCatNumber.has(catNumber)
+          );
+          const unavailableCOs = catalogNumbers.filter(
+            (catNumber) => !returnCOCatNumber.has(catNumber)
+          );
+          setAvailableCatNumbers(availableCOs);
+          setUnavailableCatNumbers(unavailableCOs);
+          // Or move the if else here
+          if (
+            unavailableCatNumbers !== undefined &&
+            unavailableCatNumbers.length > 0
+          )
+            setState({
+              type: 'UsedCatalogNumberState',
+              unavailable: unavailableCatNumbers,
+            });
+          else {
+            // Const interaction = new actionTable.Resource();
+            const interaction = new tables.Accession.Resource();
+
+            const COs = data?.map((available) => {
+              const id = available.id;
+              const co = new tables.CollectionObject.Resource({ id });
+              return co.set('accession', interaction);
+            });
+            COs.forEach((co) => {
+              interaction.set('collectionObjects', co as never);
+            });
```

Comment:
@maxpatiiuk @melton-jason 
Would you have recommendations on how we could have this piece of code functioning? 

> How to have the COs set on the Accession and/or how to have the accession set on each CO? 

This code adds a CO resource in the independent relationship of the Accession but the models stay empty

Reply:
after you done `co.set('accession', interaction);`, you need to save the changes to each CO.

`await co.save()`

doing `interaction.set('collectionObjects', co as never);` is not necessary as the accessionId field is stored in the CollectionObject table - the Accession table does not store collectionObjectIds

Reply:
thank you!